project/BuildGearpump.scala

```diff
@@ -0,0 +1,258 @@
+    gearpumpHadoop,
+    packProject
+  ) ++ examples ++ experiments ++ externals ++ integrationTests
+
+  lazy val root = Project(
+    id = "gearpump",
+    base = file("."),
+    settings = commonSettings ++ noPublish ++ gearpumpUnidocSetting,
+    aggregate = aggregated)
+    .settings(Defaults.itSettings: _*)
+    .disablePlugins(sbtassembly.AssemblyPlugin)
+
+  lazy val core = Project(
+    id = "gearpump-core",
+    base = file("core"),
+    settings = commonSettings ++ myAssemblySettings ++ javadocSettings ++ coreDependencies ++
+      addArtifact(Artifact("gearpump-core"), sbtassembly.AssemblyKeys.assembly) ++ Seq(
+      assemblyShadeRules in assembly := Seq(
+        ShadeRule.zap("akka.**").inAll,
+        ShadeRule.zap("com.google.protobuf.**").inAll,
+        ShadeRule.zap("com.typesafe.**").inAll,
+        ShadeRule.zap("net.jpountz.**").inAll,
+        ShadeRule.zap("org.apache.commons.**").inAll,
+        ShadeRule.zap("org.apache.log4j.**").inAll,
+        ShadeRule.zap("org.jboss.**").inAll,
+        ShadeRule.zap("org.objectweb.**").inAll,
+        ShadeRule.zap("org.reactivestreams.**").inAll,
+        ShadeRule.zap("org.slf4j.**").inAll,
+        ShadeRule.zap("org.uncommons.maths.**").inAll,
+        ShadeRule.zap("scala.**").inAll,
+        ShadeRule.zap("spray.**").inAll,
+        ShadeRule.rename("com.romix.**" -> "org.apache.gearpump.romix.@1").inAll,
+        ShadeRule.rename("com.esotericsoftware.**" ->
+          "org.apache.gearpump.esotericsoftware.@1").inAll,
+        ShadeRule.rename("org.objenesis.**" -> "org.apache.gearpump.objenesis.@1").inAll,
+        ShadeRule.rename("com.google.**" -> "org.apache.gearpump.google.@1").inAll,
+        ShadeRule.rename("com.codahale.metrics.**" ->
+          "org.apache.gearpump.codahale.metrics.@1").inAll
+      ),
+
+      pomPostProcess := {
+        (node: xml.Node) => changeShadedDeps(
+          Set(
+            "com.github.romix.akka",
+            "com.google.guava",
+            "com.codahale.metrics",
+            "org.scala-lang",
+            "org.scoverage"
+          ), List.empty[xml.Node], node)
+      }
+    ))
+
+  lazy val streaming = Project(
+    id = "gearpump-streaming",
+    base = file("streaming"),
+    settings = commonSettings ++ myAssemblySettings ++ javadocSettings ++
+      addArtifact(Artifact("gearpump-streaming"), sbtassembly.AssemblyKeys.assembly) ++
+      Seq(
+        assemblyShadeRules in assembly := Seq(
+          ShadeRule.rename("com.gs.collections.**" ->
+            "org.apache.gearpump.gs.collections.@1").inAll
+        ),
+
+        assemblyMergeStrategy in assembly := {
+          case "geardefault.conf" =>
+            MergeStrategy.last
+          case x =>
+            val oldStrategy = (assemblyMergeStrategy in assembly).value
+            oldStrategy(x)
+        },
+
+        libraryDependencies ++= Seq(
+          "com.goldmansachs" % "gs-collections" % gsCollectionsVersion
+        ),
+
+        pomPostProcess := {
```

Comment:
I don't get it. These shaded library are already been assembled into the gearpump-core jar file right, which means there should not be any dependency any more?